### ISSUE TITLE
[tests] error thrown if dimensions don't match

### ISSUE THREAD

**Issue by tlienart** [mateuszbaran/CovarianceEstimation.jl#22]
is this still relevant?

```julia
# XXX broken?
    # @test_throws ArgumentError cov(ce, X, dims=3)
```

**Comment by mateuszbaran** [comment]
`cov` from Statistics gives a nice error message when `dims` is smaller that 1 but an ugly one when `dims` is larger than 2. I was thinking whether we should provide a nicer error message for `dims > 2` but I didn't make up my mind so I just left a broken test.

**Comment by mateuszbaran** [comment]
Resolved by #61
